language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/util/ContextTypeResolver.java

```diff
@@ -702,6 +707,85 @@ public Optional<TypeSymbol> transform(TableConstructorExpressionNode node) {
+            if (optionalSymbol.isEmpty()) {
+                return Optional.empty();
+            }
+            typeSymbol = SymbolUtil.getTypeDescriptor(optionalSymbol.get());
+
+            if (typeSymbol.isEmpty() || typeSymbol.get().typeKind() == TypeDescKind.COMPILATION_ERROR) {
+                return Optional.empty();
+            }
+            
+            return Optional.of(buildUnionOfIterables(typeSymbol.get(), context.currentSemanticModel().get()));
+        }
+
+        // from <cursor> in iterable
+        Optional<Symbol> optionalSymbol = context.currentSemanticModel().get()
+                .symbol(fromClauseNode.typedBindingPattern().bindingPattern());
+        if (optionalSymbol.isEmpty()) {
+            Optional<TypeSymbol> typeSymbol = context.currentSemanticModel().get()
+                    .typeOf(fromClauseNode.expression());
+            if (typeSymbol.isEmpty()) {
+                return typeSymbol;
+            }
+            switch (typeSymbol.get().typeKind()) {
+                case ARRAY:
+                    return Optional.of(((ArrayTypeSymbol) typeSymbol.get()).memberTypeDescriptor());
+                case STRING:
+                    return Optional.of(context.currentSemanticModel().get().types().STRING);
+                case TABLE:
+                    return Optional.of(((TableTypeSymbol) typeSymbol.get()).rowTypeParameter());
+                case STREAM:
+                    return Optional.of(((StreamTypeSymbol) typeSymbol.get()).typeParameter());
+                case XML:
+                    return Optional.of(context.currentSemanticModel().get().types().XML);
+                case MAP:
+                    return Optional.of(((MapTypeSymbol) typeSymbol.get()).typeParam());
+            }
+        }
+        
+        return Optional.empty();
+    }
+
+    private UnionTypeSymbol buildUnionOfIterables(TypeSymbol typeSymbol, SemanticModel semanticModel) {
+        Types types = semanticModel.types();
+        TypeBuilder builder = types.builder();
+        List<TypeSymbol> unionTypeMembers = new ArrayList<>(
+                List.of(builder.ARRAY_TYPE.withType(typeSymbol).build(),
+                        builder.MAP_TYPE.withTypeParam(typeSymbol).build(),
+                        builder.STREAM_TYPE.withValueType(typeSymbol).build()));
+        if (CommonUtil.getRawType(typeSymbol).typeKind() == TypeDescKind.RECORD) {
+            try {
+                unionTypeMembers
+                        .add(builder.TABLE_TYPE.withRowType(CommonUtil.getRawType(typeSymbol)).build());
+            } catch (IllegalArgumentException ignored) {
+            }
+        }
+        if (typeSymbol instanceof StringTypeSymbol) {
+            unionTypeMembers.add(types.STRING);
+        }
+        if (typeSymbol.typeKind() == TypeDescKind.XML) {
+            unionTypeMembers.add(types.XML);
+        }
+        return builder.UNION_TYPE
```

Comment:
Have we checked if the types API throws an error if the members are empty. Let's handle that error as well.

Reply:
Since `unionTypeMembers` is never empty we can ignore the above comment